### DISCUSSION TITLE
Extend Mailer OTP Expiration

### DISCUSSION THREAD

**Discussion by eliabieri** [supabase/supabase#14335] [Feature Requests]
By default, Mailer OTPs expire after 24h.
This limit cannot be extended, which is quite annoying for my use case, as some users take several days to accept the invitation.
Is there a technical reason why this limit cannot be extended further?

**Comment by bombillazo** [comment]
Hello! We also would like to have the ability to customize and extend link expiration generated by the `generateLink` admin auth function.

**Comment by mertcorumlu** [comment]
It is really annoying that we cannot set custom expiration times for invitation, password reset links etc. I think in the self hosted version (at least) we are able to set "OTP_EXP" variable to extend link expiration, however that is not ideal as when I set OTP_EXP to 1 week, password reset links will also have a lifetime of 1 week, which I think is not ideal.

We need a more suitable way to set these parameters independently via ENV variables, or as suggested via option in `generateLink` function
